### PR TITLE
Synthetic data: adding positive data option

### PR DESCRIPTION
**Pull request by cassiofragadantas**
Adding a synthetic simulation scenario with non-negative data (X,y), controlled by flag `pos_data`.
I believe this new simulation scenario to be essential as it is commonly considered in practice.
Also, it can significantly affect the benchmark results. For instance, the fastest solver can be `cd` or `scipy` (active-set) depending on whether (X,y) is positive or not (see below).

`benchopt run -e . -d Simulated[n_features=100,n_samples=1000] -s cd -s scipy -r 1`

Screenshot from 2022-07-08 16-10-03
Screenshot from 2022-07-08 16-10-16

### CODE REVIEW

File: datasets/simulated.py [hunk 1]
```diff
@@ -11,19 +11,26 @@ class Dataset(BaseDataset):
 
     # List of parameters to generate the datasets. The benchmark will consider
     # the cross product for each key in the dictionary.
-    parameters = {'n_samples, n_features': [(100, 5000), (100, 10000)]}
+    parameters = {
+        'n_samples, n_features': [(100, 5000), (100, 10000)],
+        'pos_data': [True,False],
```
Comment (agramfort) [inline datasets/simulated.py]:
```suggestion
        'pos_data': [True, False],
```

File: datasets/simulated.py [hunk 2]
```diff
@@ -11,19 +11,26 @@ class Dataset(BaseDataset):
...
         self.n_samples = n_samples
         self.n_features = n_features
+        self.pos_data = pos_data
         self.random_state = random_state
 
     def get_data(self):
 
         rng = np.random.RandomState(self.random_state)
         X = rng.randn(self.n_samples, self.n_features)
         y = rng.randn(self.n_samples)
```
Comment (agramfort) [inline datasets/simulated.py]: to be honest I think that having X and y unrelated as random data makes little sense for a bench. I think we fixed this elsewhere by simulating a coef and doing y = X @ coef + noise.

Comment (cassiofragadantas) [reply datasets/simulated.py]: I agree. As far as I checked, we now use the function `make_correlated_data` from the benchopt main repository. The problem is it does not currently support non-negative datasets. I see two options here:
1) Adapt and use the already-existing `make_correlated_data` (requires a pull request on the main benchopt repository, but remains quite simple).
2) Do it "locally" in `simulated.py  ̀by doing y = X @ coef + noise 

Any preferences? I believe option 1 is cleaner.

**Comment by agramfort** [comment]
Yes 1) is best

**Comment by cassiofragadantas** [comment]
The current code requires benchopt/benchopt#443 (gives an error otherwise)

**Comment by cassiofragadantas** [comment]
Went for solution 2 in the end (i.e, generate positive data "locally" to avoid modifying make_correlated_data function, cf. discussion in benchopt/benchopt#443)
@agramfort, @tomMoral

**Review by tomMoral — COMMENTED** [review]
lGtm! nice one @cassiofragadantas

**Pull request merged**